packages/react-composites/src/composites/CallComposite/utils/MediaGalleryUtils.ts

```diff
@@ -0,0 +1,156 @@
+  threeParticipantJoinedNoticeString: string;
+  participantLeftNoticeString: string;
+  twoParticipantLeftNoticeString: string;
+  threeParticipantLeftNoticeString: string;
+  unnamedParticipantChangedString: string;
+  manyParticipantsChanged: string;
+  manyUnnamedParticipantsChanged: string;
+};
+
+/**
+ * sets the announcement string whenever a Participant comes or goes from a call to be
+ * used by the system narrator.
+ *
+ * @returns string to be used by the narrator and Announcer component
+ *
+ * @internal
+ */
+export const useParticipantChangedAnnouncement = (): string => {
+  const locale = useLocale().strings.call;
+  const strings = useMemo(() => {
+    return {
+      participantJoinedNoticeString: locale.participantJoinedNoticeString,
+      twoParticipantJoinedNoticeString: locale.twoParticipantJoinedNoticeString,
+      threeParticipantJoinedNoticeString: locale.threeParticipantJoinedNoticeString,
+      participantLeftNoticeString: locale.participantLeftNoticeString,
+      twoParticipantLeftNoticeString: locale.twoParticipantLeftNoticeString,
+      threeParticipantLeftNoticeString: locale.threeParticipantLeftNoticeString,
+      unnamedParticipantChangedString: locale.unnamedParticipantChangedString,
+      manyParticipantsChanged: locale.manyParticipantsChanged,
+      manyUnnamedParticipantsChanged: locale.manyUnnamedParticipantsChanged
+    };
+  }, [locale]);
+  const [announcerString, setAnnouncerString] = useState<string>('');
+  const remoteParticipants = useSelector(getRemoteParticipantsConnectedSelector);
+  const [previousParticipants, setPreviousParticipants] = useState<RemoteParticipantState[]>(remoteParticipants);
+
+  useEffect(() => {
+    const setParticipantEventString = (string: string): void => {
+      setAnnouncerString('');
+      setAnnouncerString(string);
+    };
+
+    if (previousParticipants.length > remoteParticipants.length) {
+      const whoLeft = previousParticipants.filter((p) => !remoteParticipants.includes(p));
+      //someone left
+      setParticipantEventString(createAnnouncmentString('left', whoLeft, strings));
+      setPreviousParticipants(remoteParticipants);
+    } else if (remoteParticipants.length > previousParticipants.length) {
+      const whoJoined = remoteParticipants.filter((p) => !previousParticipants.includes(p));
+      // someone joined
+      setParticipantEventString(createAnnouncmentString('joined', whoJoined, strings));
+      setPreviousParticipants(remoteParticipants);
+    }
+  }, [remoteParticipants, previousParticipants, strings]);
+
+  return announcerString;
+};
+
+/**
+ * Generates the announcement string for when a participant joins or leaves a call.
+ */
+export const createAnnouncmentString = (
+  direction: 'joined' | 'left',
+  participants: RemoteParticipantState[],
+  strings: ParticipantChangedAnnouncmentStrings
+): string => {
+  /**
+   * Check that we have more than 1 participant, if they all have no displayName return unnamed participants
```

Comment:
what is the purpose of this? doesn't your switch case below handle everything?

Reply:
It won't catch the chance that 2 unnamed participants and 1 named participant joins the call. It's probably also a rare enough case that reading "unnamed participant, unnamed participant and unnamed participant joined the call" is fine (3 people of that name _did_ join and it also lets the a11y user know that three people joined the call)

Reply:
> Should we never announce unnamed participant

I'd like a11y team to sign off this assumption. There is some value in knowing that someone without a name joined your meeting - should they even be there without a name?